topic_dynamics/parsing.py

```diff
@@ -0,0 +1,342 @@
+                subdirectory = os.path.abspath(os.path.join(directory, td, date.strftime("%Y-%m-%d")))
+                checkout_by_date(repository, subdirectory, date)
+                files = get_files(subdirectory, get_extensions(lang))
+                starting_index = count + 1
+                for file in files:
+                    if os.path.isfile(file):  # TODO: implement a better file-checking mechanism
+                        try:
+                            identifiers = get_identifiers(file, lang)
+                            if len(identifiers) != 0:
+                                count += 1
+                                formatted_identifiers = transform_identifiers(identifiers)
+                                fout.write(str(count) + ";" + os.path.relpath(file, os.path.abspath(os.path.join(directory, td))) + ";" + ",".join(formatted_identifiers) + "\n")
+                        except UnicodeDecodeError:
+                            continue
+                ending_index = count
+                dates_indices[date.strftime("%Y-%m-%d")] = (starting_index, ending_index)
+    # Write the index boundaries of slices into a separate log file
+    print("Writing the index boundaries of slices into an auxiliary file.")
+    with open(os.path.abspath(os.path.join(directory, name + "_slices.txt")), "w+") as fout:
+        for date in dates_indices.keys():
+            fout.write(date + ";" + str(dates_indices[date][0]) + "," + str(dates_indices[date][1]) + "\n")
+
+
+def split_token_file(directory: str, name: str) -> None:
+    """
+    Split a single file with tokens into splits for calculating diffs.
+    :param directory: the directory with the dataset.
+    :param name: name of the processed dataset.
+    :return: None.
+    """
+    print("Splitting the tokens of full files by versions.")
+    slice_number = 0
+    dates_indices = {}
+    os.mkdir(os.path.abspath(os.path.join(directory, name + "_slices_tokens")))
+    # Read the data about the indices boundaries of slices
+    with open(os.path.abspath(os.path.join(directory, name + "_slices.txt")), "r") as fin:
+        for line in fin:
+            slice_number = slice_number + 1
+            dates_indices[slice_number] = (int(line.rstrip().split(";")[1].split(",")[0]),
+                                           int(line.rstrip().split(";")[1].split(",")[1]))
+    # Write the tokens of each slice into a separate file, numbered incrementally
+    for date in tqdm(dates_indices.keys()):
+        with open(os.path.abspath(os.path.join(directory, name + "_tokens.txt")), "r") as fin, open(os.path.abspath(os.path.join(directory, name + "_slices_tokens", str(date) + ".txt")), "w+") as fout:
+            for line in fin:
+                if (int(line.split(";")[0]) >= dates_indices[date][0]) and (int(line.split(";")[0]) <= dates_indices[date][1]):
+                    fout.write(line)
+
+
+def calculate_diffs(directory: str, name: str, dates: List) -> None:
```

Comment:
I will review the rest - when you fix comments above - some of the comments can be easily be applied below too.